examples/point-of-sale/index.mjs

```diff
@@ -0,0 +1,59 @@
+import { loadStdlib } from "@reach-sh/stdlib";
+import * as backend from './build/index.main.mjs';
+const stdlib = loadStdlib({REACH_NO_WARN: 'Y'});
+const MIN_PRICE = 10;
+const USERS = 10;
+const FAILS = 2;
+let txns = 0;
+
+const accA = await stdlib.newTestAccount(stdlib.parseCurrency(5000));
+const ctcA = accA.contract(backend);
+const loyalTok = await stdlib.launchToken(accA, "Loyalty", "LYL", {supply: USERS});
+
+console.log(`Welcome to the point-of-sale machine. This machine processes
+payments of varying amounts and rewards each purchase with a loyalty token.
+The pos will also process refunds, which return network tokens to the customer,
+and loyalty tokens to the contract.\n`);
+
+const getPurchasePrice = (i) => {
+  const price = Math.floor(Math.random() * 100) + MIN_PRICE;
+  return (i == 0 ? 0 : price)
+};
+
+const startBuyers = async () => {
+  const runBuyer = async (i) => {
+    const acc = await stdlib.newTestAccount(stdlib.parseCurrency(100));
+    const ctc = acc.contract(backend, ctcA.getInfo());
+    const cost = getPurchasePrice(i);
+    await acc.tokenAccept(loyalTok.id);
+    try{
+      await ctc.apis.Buyer.purchase(cost);
+      txns++;
```

Comment:
This is not atomic in JS and it is possible to get the same number multiple times. You could have the contract keep track of a counter and return it, then it would be accurate.